### ISSUE TITLE
[v2] Exception in mac touchbar code while creating editor header component

### ISSUE THREAD

**Issue by jimkyndemeyer** [jimkyndemeyer/js-graphql-intellij-plugin#166]
Reported by @koresar

Bug report. I have a MacBook Pro with a touch bar.

![image](https://user-images.githubusercontent.com/1622575/42297504-3f92185e-8043-11e8-8593-8ca24e7bca92.png)

```
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'actions' of com/intellij/ui/mac/touchbar/TouchBar.buildFromGroup must not be null
	at com.intellij.ui.mac.touchbar.TouchBar.a(TouchBar.java)
	at com.intellij.ui.mac.touchbar.TouchBar.buildFromGroup(TouchBar.java)
	at com.intellij.ui.mac.touchbar.TouchBarsManager.onUpdateEditorHeader(TouchBarsManager.java:300)
	at com.intellij.openapi.editor.impl.EditorImpl.setHeaderComponent(EditorImpl.java:1817)
	at com.intellij.lang.jsgraphql.v1.ide.project.JSGraphQLLanguageUIProjectService.lambda$insertEditorHeaderComponentIfApplicable$6(JSGraphQLLanguageUIProjectService.java:292)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:3078)
	at com.intellij.lang.jsgraphql.v1.ide.project.JSGraphQLLanguageUIProjectService.insertEditorHeaderComponentIfApplicable(JSGraphQLLanguageUIProjectService.java:290)
	at com.intellij.lang.jsgraphql.v1.ide.project.JSGraphQLLanguageUIProjectService.lambda$new$0(JSGraphQLLanguageUIProjectService.java:173)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.i(IdeEventQueue.java:781)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:722)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:382)
	at com.intellij.ide.IdeEventQueue.pumpEventsForHierarchy(IdeEventQueue.java:864)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:205)
	at com.intellij.openapi.progress.util.ProgressWindow.startBlocking(ProgressWindow.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runProcessWithProgressSynchronously(ApplicationImpl.java:584)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.runProcessWithProgressSynchronously(ProgressManagerImpl.java:109)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressSynchronously(CoreProgressManager.java:214)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.c(ProjectManagerImpl.java:542)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.convertAndLoadProject(ProjectManagerImpl.java:528)
	at com.intellij.platform.PlatformProjectOpenProcessor.doOpenProject(PlatformProjectOpenProcessor.java:203)
	at com.intellij.ide.RecentProjectsManagerBase.doOpenProject(RecentProjectsManagerBase.java:579)
	at com.intellij.ide.RecentProjectsManagerBase.doReopenLastProject(RecentProjectsManagerBase.java:706)
	at com.intellij.ide.RecentProjectsManagerBase$MyAppLifecycleListener.appStarting(RecentProjectsManagerBase.java:748)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.util.messages.impl.MessageBusConnectionImpl.deliverMessage(MessageBusConnectionImpl.java:117)
	at com.intellij.util.messages.impl.MessageBusImpl.doPumpMessages(MessageBusImpl.java:426)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpWaitingBuses(MessageBusImpl.java:387)
	at com.intellij.util.messages.impl.MessageBusImpl.pumpMessages(MessageBusImpl.java:376)
	at com.intellij.util.messages.impl.MessageBusImpl.sendMessage(MessageBusImpl.java:357)
	at com.intellij.util.messages.impl.MessageBusImpl.access$200(MessageBusImpl.java:43)
	at com.intellij.util.messages.impl.MessageBusImpl$2.invoke(MessageBusImpl.java:208)
	at com.sun.proxy.$Proxy57.appStarting(Unknown Source)
	at com.intellij.idea.IdeaApplication$IdeStarter.a(IdeaApplication.java:371)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:111)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransaction(TransactionGuardImpl.java:120)
	at com.intellij.openapi.application.TransactionGuard.submitTransaction(TransactionGuard.java:122)
	at com.intellij.idea.IdeaApplication$IdeStarter.main(IdeaApplication.java:369)
	at com.intellij.idea.IdeaApplication.b(IdeaApplication.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.performUserActivity(TransactionGuardImpl.java:195)
	at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:209)
	at com.intellij.idea.IdeaApplication.a(IdeaApplication.java:72)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:762)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:732)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:361)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by jimkyndemeyer** [comment]
@artemm-bochkarev Including you on this issue based on git history of `TouchBarsManager` and `TouchBar` in intellij-community. See the stack trace above.

The setHeaderComponent method that this plugin makes use of seems to trip up a non-null in the `TouchBar` class. https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/blob/240c8d060dca1afd3d420e990cf3a338cc8b3d4d/src/main/com/intellij/lang/jsgraphql/v1/ide/project/JSGraphQLLanguageUIProjectService.java#L291-L292

Is this caused by a recent change of the touch bar UI? This plugin has been inserting a header component for a couple of years now without issue.

**Comment by koresar** [comment]
I should mention that it is reproducible in WebStorm-EAP 2018.2.
Haven't tried the plugin 2018.1 yet.

**Comment by artemm-bochkarev** [comment]
Yes, it was caused by recent changes of touchbar UI. I'll push the fix into master today.

**Comment by jimkyndemeyer** [comment]
Sounds good, thanks.

Closing this from our end.

**Comment by artemm-bochkarev** [comment]
fixed in 095fe0a "MacBook touch bar: don't create touchbar when component hasn't linked ActionGroup"
